library/src/main/java/com/novoda/downloadmanager/lib/BatchRepository.java

```diff
@@ -168,23 +163,39 @@ public DownloadBatch retrieveBatchBy(DownloadInfo downloadInfo) {
             batchesCursor.close();
         }
 
-        if (!forDeletion.isEmpty()) {
-            deleteBatchesForIds(forDeletion, downloads);
+        return batches;
+    }
+
+    public void deleteMarkedBatchesFor(Collection<DownloadInfo> downloads) {
+        Cursor batchesCursor = resolver.query(Downloads.Impl.BATCH_CONTENT_URI, null, SELECT_MARKED_FOR_DELETION, null, null);
```

Comment:
I was more inclined to extract the constant as I said earlier, but then taking a look at the resolver's `query` method in regards with the caching I went for merging it as it is. But now I see @blundell 's point about respecting the api contract so I think I might be better to comply to using the wildcard and `selectionArgs`.